visualization/app/codeCharta/codeCharta.service.ts

```diff
@@ -1,47 +1,51 @@
 import { validate } from "./util/fileValidator"
-import { AttributeTypes, CCFile, NameDataPair, BlacklistType, BlacklistItem } from "./codeCharta.model"
+import { AttributeTypes, BlacklistItem, BlacklistType, CCFile, NameDataPair } from "./codeCharta.model"
 import _ from "lodash"
 import { NodeDecorator } from "./util/nodeDecorator"
 import { ExportBlacklistType, ExportCCFile, OldAttributeTypes } from "./codeCharta.api.model"
 import { StoreService } from "./state/store.service"
-import { addFile, setSingle } from "./state/store/files/files.actions"
+import { resetFiles, setFiles, setSingle } from "./state/store/files/files.actions"
 import { getCCFiles } from "./model/files/files.helper"
 import { DialogService } from "./ui/dialog/dialog.service"
 import { setState } from "./state/store/state.actions"
 import { ScenarioHelper } from "./util/scenarioHelper"
 import { MetricService } from "./state/metric.service"
+import { setIsLoadingFile } from "./state/store/appSettings/isLoadingFile/isLoadingFile.actions"
+import { FileSelectionState, FileState } from "./model/files/files"
 
 export class CodeChartaService {
 	public static ROOT_NAME = "root"
 	public static ROOT_PATH = "/" + CodeChartaService.ROOT_NAME
 	public static readonly CC_FILE_EXTENSION = ".cc.json"
+	private fileStates: FileState[] = []
 
 	constructor(private storeService: StoreService, private dialogService: DialogService, private metricService: MetricService) {}
 
 	public loadFiles(nameDataPairs: NameDataPair[]) {
-		let printErrors = true
-
-		nameDataPairs.forEach((nameDataPair: NameDataPair) => {
-			if (!printErrors) {
-				return
-			}
-
+		for (const nameDataPair of nameDataPairs) {
 			try {
 				validate(nameDataPair.content)
 				this.addFile(nameDataPair.fileName, nameDataPair.content)
 			} catch (e) {
 				if (!_.isEmpty(e.error)) {
+					this.fileStates = []
+					this.storeService.dispatch(setIsLoadingFile(false))
 					this.dialogService.showValidationErrorDialog(e)
-					printErrors = false
+					break
 				}
 
 				if (!_.isEmpty(e.warning)) {
+					this.storeService.dispatch(setIsLoadingFile(false))
```

Comment:
Remove this line. The flag will be set to false after rendering the map. Since we always add files with warnings to our state, we can be sure that the flag will be removed.
The flag will be removed ONCE this way. Right now it's being removed after each encountered file with warnings.